source/common/access_log/access_log_formatter.cc

```diff
@@ -850,6 +857,41 @@ ResponseTrailerFormatter::formatValue(const Http::RequestHeaderMap&, const Http:
   return HeaderFormatter::formatValue(response_trailers);
 }
 
+GrpcStatusFormatter::GrpcStatusFormatter(const std::string& main_header,
+                                         const std::string& alternative_header,
+                                         absl::optional<size_t> max_length)
+    : HeaderFormatter(main_header, alternative_header, max_length) {}
+
+std::string GrpcStatusFormatter::format(const Http::RequestHeaderMap&,
+                                        const Http::ResponseHeaderMap& response_headers,
+                                        const Http::ResponseTrailerMap& response_trailers,
+                                        const StreamInfo::StreamInfo&) const {
+  std::string grpc_status_code_str = HeaderFormatter::format(response_trailers);
```

Comment:
Not being a directly related problem, It seems that `getGrpcStatus` can only extract grpc-status from response trailers. I think that it should be enabled to extract it from the response header because it has a possibility to include it in response header descrived above comment.

Reply:
And, I think that we should use `HeaderFormatter::format` because `"grpc-status"` has registered to `HeaderFormatter` already and it can return valid value like `UnspecifiedValueString` in logging region.

Reply:
It is pretty inefficient that we format grpc-status to string and parse it then convert to string again. I think we should just log it as a number. Reusing HeaderFormatter seems unnecessary.

There is a version of `getGrpcStatus` takes both trailers and headers, use it like this: https://github.com/envoyproxy/envoy/blob/master/source/common/access_log/access_log_impl.cc#L242